src/Server/HTTPHandlerFactory.cpp

```diff
@@ -26,6 +26,42 @@ namespace ErrorCodes
     extern const int INVALID_CONFIG_PARAMETER;
 }
 
+namespace
+{
+
+class RedirectRequestHandler : public HTTPRequestHandler
+{
+private:
+    std::string url;
+
+public:
+    explicit RedirectRequestHandler(std::string url_)
+        : url(std::move(url_))
+    {
+    }
+
+    void handleRequest(HTTPServerRequest &, HTTPServerResponse & response, const ProfileEvents::Event &) override
+    {
+        response.redirect(url);
+    }
+};
+
+HTTPRequestHandlerFactoryPtr createRedirectHandlerFactory(
+    const Poco::Util::AbstractConfiguration & config,
+    const std::string & config_prefix)
+{
+    std::string url = config.getString(config_prefix + ".handler.location");
+
+    auto factory = std::make_shared<HandlingRuleHTTPHandlerFactory<RedirectRequestHandler>>(
+        [my_url = std::move(url)]() mutable { return std::make_unique<RedirectRequestHandler>(std::move(my_url)); });
```

Comment:
we shouldn't move from capture - `RedirectRequestHandler` constructor will be called repeatedly on each matching request, and capture will be cleaned with `std::move()` on the first 
```suggestion
        [my_url = std::move(url)]() { return std::make_unique<RedirectRequestHandler>(my_url); });
```